pkg/archives/archives.go

```diff
@@ -179,3 +179,39 @@ func IsTGZ(b []byte) bool {
 	_, err = tarReader.Next()
 	return err == nil
 }
+
+func CreateFilteredAirgapBundle(airgapBundle string, filesToInclude []string) (io.Reader, error) {
+	buf := bytes.NewBuffer(nil)
+	gw := gzip.NewWriter(buf)
+	tw := tar.NewWriter(gw)
+
+	for _, path := range filesToInclude {
+		contents, err := GetFileFromAirgap(path, airgapBundle)
```

Comment:
an optimization here would be to iterate over the bundle only once and write desired files to disk directly rather than loading each file in memory and having to iterate over the bundle files for each file.